### ISSUE TITLE
Why check 'ListSets' repeatedly when not supported by provider?

### ISSUE THREAD

**Issue by twagoo** [clarin-eric/oai-harvest-manager#10]
I noted that in the harvesting of [SIL](http://archive.ethnologue.com/sil_archives.asp), the log file shows the following pattern:
```
15:00:19,953 INFO [Thread-13] RecordHarvesting - Fetched olac record oai:sil.org:9721
15:00:19,958 WARN [Thread-13] TransformAction - Transformer warning: WRN: ListsSets[http://archive.ethnologue.com/sil_archives.asp?verb=ListSets] not available
15:00:23,171 INFO [Thread-13] RecordHarvesting - Fetched olac record oai:sil.org:9726
15:00:23,185 WARN [Thread-13] TransformAction - Transformer warning: WRN: ListsSets[http://archive.ethnologue.com/sil_archives.asp?verb=ListSets] not available
15:00:26,389 INFO [Thread-13] RecordHarvesting - Fetched olac record oai:sil.org:9727
15:00:26,394 WARN [Thread-13] TransformAction - Transformer warning: WRN: ListsSets[http://archive.ethnologue.com/sil_archives.asp?verb=ListSets] not available
...
```

It seems that between every fetched record, a call to [?verb=ListSets](http://archive.ethnologue.com/sil_archives.asp?verb=ListSets) is made, which has the following response:
```xml
<error code="noSetHierarchy">This repository does not support sets.</error>
```
Given this response, it seems to me that all checks but the first one of this kind are redundant. 

Would it be possible to adapt the harvester and make it a bit smarter in this regard? My gut feeling is that it could save a lot of time, especially with slower and/or large providers such as SIL.

**Comment by menzowindhouwer** [comment]
Ah, this is part of the https://github.com/clarin-eric/oai-harvest-manager/blob/master/src/main/resources/config-others.xml#L54 action. The transform is able to cache these kind of requests, but it looks like SIL escapes from it ...

**Comment by menzowindhouwer** [comment]
Two problems here:
- the requests mentioned in the OAI envelope is wrong and when used as the basis for the ListSets leads to a 404
- when using the right URI java.net.URI still receives a 403, while the URI resolved nicely in the browser, i.e., this endpoint wants another user.agent
When both problems are solved the ListSets response is properly cached.

**Comment by menzowindhouwer** [comment]
closed by d7a5433
